classes/sota.bbclass

```diff
@@ -0,0 +1,65 @@
+DISTROOVERRIDES .= "${@bb.utils.contains('DISTRO_FEATURES', 'sota', ':sota', '', d)}"
+
+SOTA_CLIENT_PROV ??= "aktualizr-shared-prov"
+SOTA_DEPLOY_CREDENTIALS ?= "1"
+SOTA_HARDWARE_ID ??= "${MACHINE}"
+
+IMAGE_CLASSES += " image_types_ostree image_types_ota image_repo_manifest"
+IMAGE_INSTALL:append:sota = " aktualizr aktualizr-info ${SOTA_CLIENT_PROV} \
+                              ostree os-release ostree-kernel ostree-initramfs \
+                              ${@'ostree-devicetrees' if oe.types.boolean('${OSTREE_DEPLOY_DEVICETREE}') else ''}"
+
+IMAGE_FSTYPES += "${@bb.utils.contains('DISTRO_FEATURES', 'sota', 'ostreepush garagesign garagecheck ota-ext4', ' ', d)}"
+IMAGE_FSTYPES += "${@bb.utils.contains('BUILD_OSTREE_TARBALL', '1', 'ostree.tar.bz2', ' ', d)}"
+IMAGE_FSTYPES += "${@bb.utils.contains('BUILD_OTA_TARBALL', '1', 'ota.tar.xz', ' ', d)}"
+
+WKS_FILE:sota ?= "sdimage-sota.wks"
+
+EXTRA_IMAGEDEPENDS:append:sota = " parted-native mtools-native dosfstools-native"
+
+INITRAMFS_FSTYPES ?= "${@oe.utils.ifelse(d.getVar('OSTREE_BOOTLOADER') == 'u-boot', 'cpio.gz.u-boot', 'cpio.gz')}"
+
+# Please redefine OSTREE_REPO in order to have a persistent OSTree repo
+OSTREE_REPO ?= "${DEPLOY_DIR_IMAGE}/ostree_repo"
+OSTREE_BRANCHNAME ?= "${SOTA_HARDWARE_ID}"
+OSTREE_OSNAME ?= "poky"
+OSTREE_BOOTLOADER ??= 'u-boot'
+OSTREE_BOOT_PARTITION ??= "/boot"
+OSTREE_KERNEL ??= "${KERNEL_IMAGETYPE}"
+OSTREE_DEPLOY_DEVICETREE ??= "0"
+OSTREE_DEVICETREE ??= "${KERNEL_DEVICETREE}"
+OSTREE_MULTI_DEVICETREE_SUPPORT ??= "0"
+OSTREE_SYSROOT_READONLY ??= "0"
+OSTREE_REPO_CONFIG ?= ""
+OSTREE_OTA_REPO_CONFIG ?= ""
+
+INITRAMFS_IMAGE ?= "initramfs-ostree-image"
+
+GARAGE_SIGN_TOOL ?= "garage-sign"
```

Comment:
This is the only addition vs the meta-updater copy, we might as well upstream it directly

Reply:
Not actually. I ended up creating a PR to meta updater, and there you can find the exact diff of the changes.
They're simple enough, just replacing the hardcoded tool name inside a task in classes/image_types_ostree.bbclass and then creating the tool variable name inside classes/sota.bbclass.

https://github.com/uptane/meta-updater/pull/104

Reply:
Approved, thank you!